src/mesh/Mesh.hpp

```diff
@@ -218,7 +219,59 @@ class Mesh {
     return _communicationMap;
   }
 
-  void addMesh(Mesh &deltaMesh);
+  template <typename UnaryPredicate>
+  void filterAndAddMesh(Mesh const &other, UnaryPredicate filter)
+  {
+    PRECICE_TRACE();
+    PRECICE_ASSERT(_dimensions == other.getDimensions());
+
+    boost::container::flat_map<VertexID, Vertex *> vertexMap;
+    vertexMap.reserve(other.vertices().size());
+    Eigen::VectorXd coords(_dimensions);
+    for (const Vertex &vertex : other.vertices()) {
+      if (filter(vertex)) {
+        coords    = vertex.getCoords();
+        Vertex &v = createVertex(coords);
+        v.setGlobalIndex(vertex.getGlobalIndex());
+        if (vertex.isTagged())
+          v.tag();
+        v.setOwner(vertex.isOwner());
+        PRECICE_ASSERT(vertex.getID() >= 0, vertex.getID());
+        vertexMap[vertex.getID()] = &v;
+      }
+    }
```

Comment:
Not sure what you mean here, or rather how breaking out that part (which I did) makes a difference. I mean we don't know exactly how much space it would need either way, or was that not the point?